cupyx/scipy/sparse/sputils.py

```diff
@@ -115,3 +117,54 @@ def upcast(*args):
             return t
 
     raise TypeError('no supported conversion for types: %r' % (args,))
+
+
+def check_shape(args, current_shape=None):
+    """Check validity of the shape"""
+
+    if len(args) == 0:
+        raise TypeError("function missing 1 required positional argument: "
+                        "'shape'")
+
+    elif len(args) == 1:
+        try:
+            shape_iter = iter(args[0])
+        except TypeError:
+            new_shape = (operator.index(args[0]), )
+        else:
+            new_shape = tuple(operator.index(arg) for arg in shape_iter)
+    else:
+        new_shape = tuple(operator.index(arg) for arg in args)
+
+    if current_shape is None:
+        if len(new_shape) != 2:
+            raise ValueError('shape must be a 2-tuple of positive integers')
+        elif new_shape[0] < 0 or new_shape[1] < 0:
+            raise ValueError("'shape' elements cannot be negative")
+
+    else:
+        current_size = cupy.prod(numpy.asarray(current_shape))
```

Comment:
With your fix, the line computes `current_size` on the host. You can simply write as below (almost the same with SciPy's):
```suggestion
        current_size = numpy.prod(current_shape)
```

Note:
`cupy.prod(numpy.asarray(current_shape))` happens to call `numpy.ndarray.prod` [here](https://github.com/cupy/cupy/blob/30934f0b7a9ece99d631302316a8fba33b1805c3/cupy/_math/sumprod.py#L70), but I don't think it is a desired way to pass it a `numpy.ndarray` and `a` should be checked `cupy.ndarray` as commented there.

Reply:
Yeah you are right. I missed this part.:confused: